llvm/lib/LTO/ThinLTOCodeGenerator.cpp

```diff
@@ -448,6 +448,9 @@ class FileModuleCacheEntry : public ModuleCacheEntry {
 
   Error writeObject(const MemoryBuffer &OutputBuffer,
                     StringRef OutputPath) final {
+    // Clear output file if exists for hard-linking.
+    if (sys::fs::exists(OutputPath))
+      sys::fs::remove(OutputPath);
```

Comment:
`sys::fs::remove` declaration is

```
/// @returns errc::success if path has been removed or didn't exist, otherwise a
///          platform-specific error code. If IgnoreNonExisting is false, also
///          returns error if the file didn't exist.
std::error_code remove(const Twine &path, bool IgnoreNonExisting = true);
```

so it should be fine to just call `sys::fs::remove(OutputPath)` without the `sys::fs::exists` check.